db.go

```diff
@@ -383,11 +383,15 @@ func (db *DB) close() error {
 	if !db.opened {
 		return nil
 	}
-	
+
 	db.opened = false
 
 	db.freelist = nil
-	db.path = ""
+
+	// defer clearing the path because funlock may need it.
+	defer func() {
+		db.path = ""
+	}()
```

Comment:
Is it OK to leave the field populated if one of the error return paths is taken? What invariant is implied by setting db.path = ""?